test/spec/sanity-checks.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-env mocha */
+
+var request = require('supertest')
+var assert = require('assert')
+var path = require('path')
+var fs = require('fs')
+
+/**
+ * Basic sanity checks on the dev server
+ */
+describe('The prototype kit', function () {
+  var app
+  var server
+
+  before(function () {
+    app = require('../../server')
+    server = app.listen(3000)
```

Comment:
@joelanman Was that not before you stopped the server.js from doing the listening and put it in run-server? If you try removing that listen, you should see it still works and also stops ok. It seemed to when I tried it a moment ago. To quote supertest:

> You may pass an http.Server, or a Function to request() - if the server is not already listening for connections then it is bound to an ephemeral port for you so there is no need to keep track of ports.